minimal_data_sharing/minimal_lock_free_queue.cpp

```diff
@@ -0,0 +1,144 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <chrono>
+#include <memory>
+
+#include "rclcpp/rclcpp.hpp"
+#include "trajectory_msgs/msg/joint_trajectory.hpp"
+
+#include <iceoryx_utils/concurrent/lockfree_queue.hpp>
+
+
+using namespace std::chrono_literals;
+
+class TrajectoryGenerator : public rclcpp::Node
+{
+public:
+  TrajectoryGenerator()
+  : Node("trajectory_generator"), count_(0)
+  {
+    constexpr std::size_t TRAJECTORY_SIZE = 10;
+    publisher_ = this->create_publisher<trajectory_msgs::msg::JointTrajectory>("trajectory", 10);
+    auto timer_callback =
+      [this]() -> void {
+        auto message = trajectory_msgs::msg::JointTrajectory();
+        message.points.reserve(TRAJECTORY_SIZE);
+        for (std::size_t i = 0; i < TRAJECTORY_SIZE; i++) {
+          trajectory_msgs::msg::JointTrajectoryPoint point;
+          point.positions.push_back(count_++);
+          message.points.push_back(point);
+        }
+        RCLCPP_INFO(
+          this->get_logger(), "Publish trajectory with %zu points",
+          message.points.size());
+        this->publisher_->publish(message);
+      };
+    timer_ = this->create_wall_timer(1s, timer_callback);
+  }
+
+private:
+  rclcpp::TimerBase::SharedPtr timer_;
+  rclcpp::Publisher<trajectory_msgs::msg::JointTrajectory>::SharedPtr publisher_;
+  size_t count_;
+};
+
+class TrajectoryController : public rclcpp::Node
+{
+public:
+  using JointTrajectoryPointQueue =
+    iox::concurrent::LockFreeQueue<trajectory_msgs::msg::JointTrajectoryPoint, 100>;
+
+  TrajectoryController()
+  : Node("trajectory_controller")
+  {
+    // Create a callback function for when messages are received.
+    // Variations of this function also exist using, for example UniquePtr for zero-copy transport.
+    setvbuf(stdout, NULL, _IONBF, BUFSIZ);
+    subscription_ = this->create_subscription<trajectory_msgs::msg::JointTrajectory>(
+      "trajectory",
+      10,
+      [this](trajectory_msgs::msg::JointTrajectory::UniquePtr msg) {
+        for (const auto & point : msg->points) {
+          queue_.tryPush(point);
+        }
+        RCLCPP_INFO(this->get_logger(), "Received trajectory with %zu points ", msg->points.size());
+      });
+
+    auto timer_callback =
+      [this]() -> void {
+        auto msg = this->queue_.pop();
+        while (msg.has_value()) {
+          execute_trajectory_point(msg.value());
+          msg = this->queue_.pop();
+        }
+      };
+    realtime_callback_group_ = this->create_callback_group(
+      rclcpp::CallbackGroupType::MutuallyExclusive, false);
+
+    timer_ = this->create_wall_timer(100ms, timer_callback, realtime_callback_group_);
+  }
+
+  void execute_trajectory_point(const trajectory_msgs::msg::JointTrajectoryPoint & point)
+  {
+    // just log positions for demo purposes
```

Comment:
```suggestion
    // Logging is not safe for RT but this is just a demo for message passing
```

Maybe a bit more clear this way?

Reply:
Much better, thanks